### PR TITLE
Feature/teacher waiting list

### LINKED ISSUE #133: Make it so that when a teacher clicks on their own project, they see a list of applicants.

### PR DESCRIPTION
**Pull request by arnauds5**
There is a new field on project : "authorID", that contains the ID of the user who upload the project (teacher/TA/etc).
Only the author of a project can view/click on the menu button to see the list of waiting students.
If you need to test, you can modify the authorID of any project on Firebase (or juste upload new one).
You can find your user ID on Firebase -> Authentication.

### CODE REVIEW

**Review by kaourintin — DISMISSED** [review]
Reviewed entirely with the coder of this code.

**Review by kaourintin — APPROVED** [review]
ids updating wrongly on firebase were sucessfully fixed.

File: app/src/main/java/com/sdp13epfl2021/projmag/database/FirebaseCandidatureDatabase.kt [hunk 1]
```diff
@@ -98,9 +98,13 @@ class FirebaseCandidatureDatabase(
         getDoc(projectID)
             .get()
             .addOnSuccessListener { doc ->
-                doc?.data?.let {
-                    onSuccess(buildCandidature(it, projectID))
-                } ?: onFailure(Exception("Candidature document not found with projectID : $projectID"))
+                if (doc.exists()) {
+                    doc?.data?.let {
+                        onSuccess(buildCandidature(it, projectID))
+                    } ?: onFailure(Exception("Candidature document invalid for projectID : $projectID"))
```
Comment (lpatel99) [inline app/src/main/java/com/sdp13epfl2021/projmag/database/FirebaseCandidatureDatabase.kt:104]: Maybe avoid using a raw literal

**Review by lpatel99 — APPROVED** [review]
Nothing else to say. Works well.

**Pull request merged**